framework/integration/test/reconciliation/reconciliation_test.go

```diff
@@ -3,19 +3,165 @@
 package reconciliation
 
 import (
+	"reflect"
 	"testing"
+	"time"
 
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/operatorkit/framework"
 	"github.com/giantswarm/operatorkit/framework/integration/client"
+	"github.com/giantswarm/operatorkit/framework/integration/testresource"
 )
 
 // Test_Finalizer_Integration_Reconciliation is a integration test for
 // the proper replay and reconciliation of delete events with finalizers.
 func Test_Finalizer_Integration_Reconciliation(t *testing.T) {
+	configMapName := "test-cm"
+	testFinalizer := "operatorkit.giantswarm.io/test-operator"
 	testNamespace := "finalizer-integration-reconciliation-test"
+	testOtherFinalizer := "operatorkit.giantswarm.io/other-operator"
+	operatorName := "test-operator"
 
 	client.MustSetup(testNamespace)
 	defer client.MustTeardown(testNamespace)
+	var err error
+	var tr *testresource.Resource
+	{
+		c := testresource.Config{}
+
+		tr, err = testresource.New(c)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	resources := []framework.Resource{
+		framework.Resource(tr),
+	}
+
+	c := client.Config{
+		Resources: resources,
+
+		Name:      operatorName,
+		Namespace: testNamespace,
+	}
+
+	operatorkitFramework, err := client.NewFramework(c)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	// We create a configmap, but add a finalizer of another operator. This will
+	// cause the configmap to continue existing after the framework removes it own
+	// finalizer.
+	cm := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: testNamespace,
+			Finalizers: []string{
+				testOtherFinalizer,
+			},
+		},
+		Data: map[string]string{},
+	}
+	_, err = client.CreateConfigMap(cm)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	// We start the framework.
+	go operatorkitFramework.Boot()
+
+	// We wait the absolute maximum amount of time here:
+	// 20 second ResyncPeriod + 2 second RateWait + 3 second for safety.
+	// The framework should now add the finalizer and EnsureCreated should be hit
+	// once immediatly.
+	// EnsureCreated: 1, EnsureDeleted: 0
+	// The framework should reconcile twice in this period.
+	// EnsureCreated: 3, EnsureDeleted: 0
```

Comment:
It is super hard for me to read this. Can we maybe format the comment blog a bit to help the reader? 

```
// We wait the absolute maximum amount of time here. 20 second ResyncPeriod + 2
// second RateWait + 3 second for safety. The framework should now add the
// finalizer and EnsureCreated should be hit once immediatly.
//
//     EnsureCreated: 1, EnsureDeleted: 0
//
// The framework should reconcile twice in this period.
//
//     EnsureCreated: 3, EnsureDeleted: 0
//
```

Reply:
Done, I think it helps somewhat.